logic/src/commonMain/kotlin/com/wire/kalium/logic/data/notification/LocalNotificationMessageMapper.kt

```diff
@@ -103,4 +111,21 @@ class LocalNotificationMessageMapperImpl : LocalNotificationMessageMapper {
             messageContent.editMessageId,
             LocalNotificationUpdateMessageAction.Edit(messageContent.newContent, message.id)
         )
+
+    override fun fromEntitiesToLocalNotifications(
+        list: List<NotificationMessageEntity>,
+        messageSizePerConversation: Int,
+        mapMessage: (NotificationMessageEntity) -> LocalNotificationMessage?
+    ): List<LocalNotification.Conversation> =
+        list.groupBy { it.conversationId }
+            .map { (conversationId, messages) ->
+                LocalNotification.Conversation(
+                    // todo: needs some clean up!
+                    id = conversationId.toModel(),
+                    conversationName = messages.first().conversationName ?: "",
+                    messages = messages.take(messageSizePerConversation).mapNotNull { mapMessage(it) },
+                    isOneToOneConversation = messages.first().conversationType == ConversationEntity.Type.ONE_ON_ONE,
+                    isReplyAllowed = messages.first().degradedConversationNotified
```

Comment:
question: so if the conversation degraded we allow the reply? based on the logic I see here I think it should be the opposite?

Reply:
No, we allow reply when user was notified about degrading :) 
`degradedConversationNotified = true` means either user was notified about degraded conversation OR conversation is Verified.